ex03.ex

```diff
@@ -59,7 +59,21 @@ defmodule Ex03 do
 
   """
 
-  def odd_even . . . "your code"
```

Comment:
# -3

There are several problems here. First, it doesn't work if given an empty list. But, second, there's way too much conditional logic. My intent in this class is to develop your feeling for this, so alarm bells start ringing in your head when you see all these if statements and duplicated code.

This would be a lot better if you started by handling the empty list case specifically. This would move an entire level of nested ifs in your code.

~~~ elixir
def odd_even([]), do: []

def odd_even([head | tail]) do
  if Integer.is_odd(head) do
    [:odd | odd_even(tail)]
  else
    [:even | odd_even(tail)]
  end
end
~~~

Then you could eliminate the second `if` by moving the body into a separate function.